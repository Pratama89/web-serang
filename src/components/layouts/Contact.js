import SectionHeaders from "./SectionHeader";

export default function Contact() {
  return (
    <section className="text-center my-16  pt-24" id="contact">
      <SectionHeaders
        subHeader="Kontak Kami"
        mainHeader="Komp. Bangun Reksa Indah II Blok V No.18"
      />
      <div className="mt-16 lg:mt-20">
        <div className="grid grid-cols-1 md:grid-cols-2 gap-8">
          <div className="rounded-lg overflow-hidden">
            <iframe
              src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d991.5927426770388!2d106.703291!3d-6.21471!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x2e69f9f3a3b5d1d3:0xbcdf79510b5524e5!2sJl. Btn Blok V No.18, RT.007/RW.006, Pd. Pucung, Kec. Karang Tengah, Kota Tangerang, Banten 15157!5e0!3m2!1sid!2sid!4v1701757678280!5m2!1sid!2sid"
              width="600"
              height="450"
              style={{ border: 0 }}
              loading="lazy"
              referrerPolicy="no-referrer-when-downgrade"></iframe>
          </div>
          <div>
            <div className="max-w-full mx-auto rounded-lg overflow-hidden">
              <div className="px-6 py-4">
                <h3 className="text-lg font-medium text-gray-900">
                  Detail Alamat
                </h3>
                <p className="mt-1 text-gray-600">
                  Jl. Btn, Komp. Bangun Reksa Indah II Blok V No.18,
                  RT.007/RW.006, Pd. Pucung, Kec. Karang Tengah, Kota Tangerang,
                  Banten 15157
                </p>
              </div>
              <div className="border-t border-gray-200 px-6 py-4">
                <h3 className="text-lg font-medium text-gray-900">Jam Kerja</h3>
                <p className="mt-1 text-gray-600">
                  Senin - Jumat: 07.30 - 17.00
                </p>
                <p className="mt-1 text-gray-600">Sabtu: 07.30 - 16.00</p>
                <p className="mt-1 text-gray-600">Minggu: Tutup</p>
              </div>
              <div className="border-t border-gray-200 px-6 py-4">
                <h3 className="text-lg font-medium text-gray-900">Kontak</h3>
                <p className="mt-1 text-gray-600">
                  Email: admin@mutiaralaminating.com
                </p>
                <p className="mt-1 text-gray-600">
                  WhatsApp: +62 812-1786-9431
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
}
