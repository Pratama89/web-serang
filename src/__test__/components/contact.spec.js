import Contact from "@/components/layouts/Contact";
import { render, screen } from "@testing-library/react";
import { describe } from "node:test";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/contact",
      pathname: "",
      query: "",
      asPath: "",
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
      isReady: true,
    };
  },
}));

describe("Contact", () => {
  it("should render", () => {
    const page = render(<Contact />);
    // expect(screen.getByTestId("title").textContent).toBe("Header");
    expect(page).toMatchSnapshot();
  });
});
