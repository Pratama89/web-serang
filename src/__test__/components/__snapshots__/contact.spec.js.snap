// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="text-center my-16 mt-40"
        id="contact"
      >
        <div>
          <h3
            class="capitalize text-primary font-semibold text-3xl"
          >
            Kontak Kami
          </h3>
          <h2
            class="text-base text-gray-600 font-normal italic mt-2 md:text-sm lg:text-lg"
          >
            Komp. Bangun Reksa Indah II Blok V No.18
          </h2>
        </div>
        <div
          class="mt-16 lg:mt-20"
        >
          <div
            class="grid grid-cols-1 md:grid-cols-2 gap-8"
          >
            <div
              class="rounded-lg overflow-hidden"
            >
              <iframe
                height="450"
                loading="lazy"
                referrerpolicy="no-referrer-when-downgrade"
                src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d991.5927426770388!2d106.703291!3d-6.21471!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x2e69f9f3a3b5d1d3:0xbcdf79510b5524e5!2sJl. Btn Blok V No.18, RT.007/RW.006, Pd. Pucung, Kec. Karang Tengah, Kota Tangerang, Banten 15157!5e0!3m2!1sid!2sid!4v1701757678280!5m2!1sid!2sid"
                style="border: 0px;"
                width="600"
              />
            </div>
            <div>
              <div
                class="max-w-full mx-auto rounded-lg overflow-hidden"
              >
                <div
                  class="px-6 py-4"
                >
                  <h3
                    class="text-lg font-medium text-gray-900"
                  >
                    Detail Alamat
                  </h3>
                  <p
                    class="mt-1 text-gray-600"
                  >
                    Jl. Btn, Komp. Bangun Reksa Indah II Blok V No.18, RT.007/RW.006, Pd. Pucung, Kec. Karang Tengah, Kota Tangerang, Banten 15157
                  </p>
                </div>
                <div
                  class="border-t border-gray-200 px-6 py-4"
                >
                  <h3
                    class="text-lg font-medium text-gray-900"
                  >
                    Jam Kerja
                  </h3>
                  <p
                    class="mt-1 text-gray-600"
                  >
                    Senin - Jumat: 07.30 - 17.00
                  </p>
                  <p
                    class="mt-1 text-gray-600"
                  >
                    Sabtu: 07.30 - 16.00
                  </p>
                  <p
                    class="mt-1 text-gray-600"
                  >
                    Minggu: Tutup
                  </p>
                </div>
                <div
                  class="border-t border-gray-200 px-6 py-4"
                >
                  <h3
                    class="text-lg font-medium text-gray-900"
                  >
                    Kontak
                  </h3>
                  <p
                    class="mt-1 text-gray-600"
                  >
                    Email: admin@mutiaralaminating.com
                  </p>
                  <p
                    class="mt-1 text-gray-600"
                  >
                    WhatsApp: +62 812-1786-9431
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="text-center my-16 mt-40"
      id="contact"
    >
      <div>
        <h3
          class="capitalize text-primary font-semibold text-3xl"
        >
          Kontak Kami
        </h3>
        <h2
          class="text-base text-gray-600 font-normal italic mt-2 md:text-sm lg:text-lg"
        >
          Komp. Bangun Reksa Indah II Blok V No.18
        </h2>
      </div>
      <div
        class="mt-16 lg:mt-20"
      >
        <div
          class="grid grid-cols-1 md:grid-cols-2 gap-8"
        >
          <div
            class="rounded-lg overflow-hidden"
          >
            <iframe
              height="450"
              loading="lazy"
              referrerpolicy="no-referrer-when-downgrade"
              src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d991.5927426770388!2d106.703291!3d-6.21471!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x2e69f9f3a3b5d1d3:0xbcdf79510b5524e5!2sJl. Btn Blok V No.18, RT.007/RW.006, Pd. Pucung, Kec. Karang Tengah, Kota Tangerang, Banten 15157!5e0!3m2!1sid!2sid!4v1701757678280!5m2!1sid!2sid"
              style="border: 0px;"
              width="600"
            />
          </div>
          <div>
            <div
              class="max-w-full mx-auto rounded-lg overflow-hidden"
            >
              <div
                class="px-6 py-4"
              >
                <h3
                  class="text-lg font-medium text-gray-900"
                >
                  Detail Alamat
                </h3>
                <p
                  class="mt-1 text-gray-600"
                >
                  Jl. Btn, Komp. Bangun Reksa Indah II Blok V No.18, RT.007/RW.006, Pd. Pucung, Kec. Karang Tengah, Kota Tangerang, Banten 15157
                </p>
              </div>
              <div
                class="border-t border-gray-200 px-6 py-4"
              >
                <h3
                  class="text-lg font-medium text-gray-900"
                >
                  Jam Kerja
                </h3>
                <p
                  class="mt-1 text-gray-600"
                >
                  Senin - Jumat: 07.30 - 17.00
                </p>
                <p
                  class="mt-1 text-gray-600"
                >
                  Sabtu: 07.30 - 16.00
                </p>
                <p
                  class="mt-1 text-gray-600"
                >
                  Minggu: Tutup
                </p>
              </div>
              <div
                class="border-t border-gray-200 px-6 py-4"
              >
                <h3
                  class="text-lg font-medium text-gray-900"
                >
                  Kontak
                </h3>
                <p
                  class="mt-1 text-gray-600"
                >
                  Email: admin@mutiaralaminating.com
                </p>
                <p
                  class="mt-1 text-gray-600"
                >
                  WhatsApp: +62 812-1786-9431
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
